editor/sidebar/block-inspector/index.js

```diff
@@ -18,36 +19,76 @@ import './style.scss';
 import { deselectBlock } from '../../actions';
 import { getSelectedBlock } from '../../selectors';
 
-const BlockInspector = ( { selectedBlock, ...props } ) => {
-	if ( ! selectedBlock ) {
-		return null;
+class BlockInspector extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.forceUpdateIfSelected = this.forceUpdateIfSelected.bind( this );
+	}
+
+	componentWillMount() {
+		const { manager: rsfManager } = this.context;
+		rsfManager.onComponentsChange( 'Inspector.Controls', this.forceUpdateIfSelected );
+	}
+
+	componentWillUnmount() {
+		const { manager: rsfManager } = this.context;
+		rsfManager.removeOnComponentsChange( 'Inspector.Controls', this.forceUpdateIfSelected );
+	}
+
+	forceUpdateIfSelected() {
+		// We must bind to the React-Slot-Fill Provider manager to re-render if
+		// slots change after the initial render. This can occur when focus
+		// changes between blocks, where control fills may only update after
+		// change in selected block (and subsequent render) already occurred.
+		if ( this.props.selectedBlock ) {
+			this.forceUpdate();
```

Comment:
What's the effective difference? If we're saying we want a new `render` to occur, `forceUpdate` seems the most appropriate way to get there. Also avoids unnecessarily filtering which likely occurs in calling `getFillsByName`.

Alternatively, I could imagine an implementation where fill count is tracked in state. But this  encroaches on territory of duplicating source of truth in state, typically discouraged.